### ISSUE TITLE
Is this project dead?

### ISSUE THREAD

**Issue by wyan** [zerotier/zeronsd#233]
Instructions for Linux and latest published package rely on an obsolete libssl1.1, and there seems to be no accepted pull requests or commits for at least a year or two. Has zerotier dropped DNS support?

**Comment by bruteForce93** [comment]
Yeah, sadly It seems so. This, or zeronsd works as intended and the project has been completed. The latest 0.5.2 packages (https://github.com/zerotier/zeronsd/releases/tag/v0.5.2) fixes an inconsistency with Ubuntu's libssl1.1 and now uses libssl3 instead. But it just doesn't work out of the box for self-hosted controllers (at least not for my stupid friend over here) haha...

**Comment by JDPerf** [comment]
Maybe they're focused on [integrating this all into 2.0](https://github.com/zerotier/zeronsd/blob/main/docs/quickstart.md#status) but with a whole new code base. Either way, hopefully they catch-up on our issues/PRs and apply those contributions toward whatever the future is.

**Comment by wyan** [comment]
The line claiming it is being integrated into 2.0 "soon" is 3 years old. This doesn't fill me with confidence about the product's future...

**Comment by bruteForce93** [comment]
Well if you think about it, you already have a nice free product to implement on your own and it actually works on bare cheap metal or VM! So even if the project has been dropped you still have a nice secure VPN solution for some basic personal networking needs. My take on this is 2.0 won't be released any soon (if at all), but feel free to wish me that I am wrong.

**Comment by np22-jpg** [comment]
> The line claiming it is being integrated into 2.0 "soon" is 3 years old. This doesn't fill me with confidence about the product's future...

FWIW, there was actually a lot of behind-the-scenes stuff happening for this project as a whole. If you take a look at erikh’s [LinkedIn](https://www.linkedin.com/in/erik-hollensbe/), there’s a bunch of stuff they were apparently playing with. In addition, the devs are quite busy. Just look at the amount of stuff someara is apparently putting into private repos, hopefully the now-private `tetanus` branch of ZeroTierOne. That being said, the last report I’ve seen on 2.0 is on [yellowsite](https://news.ycombinator.com/item?id=31685741), and I have no hopes of any “2.0” release coming out anytime soon, especially with as many times as “right around the corner” has been said. Though there’s been a lot of new stuff getting released to the public, the entirety of ZeroTierOne itself has been rewritten as least 3 times, from C++ -> Go -> Rust.

For this repo specifically, it [doesn’t look like the dns lib it relies on will be updated anytime soon](https://github.com/zerotier/zeronsd/pull/221) (in fact, trust-dns is ~3 releases ahead and is now called something else entirely). Because of this, I’d be hesitant to rely on this, which is a shame, because I thought it was pretty cool. For now, I’ve just been manually creating custom DNS records on my DNS server.
